driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -400,11 +400,9 @@ public void openAsync(final SingleResultCallback<Void> callback) {
         public void close() {
             // All but the first call is a no-op
             if (!isClosed.getAndSet(true)) {
-                if (!DefaultConnectionPool.this.closed) {
-                    connectionPoolListener.connectionCheckedIn(new ConnectionCheckedInEvent(getId(wrapped)));
-                    if (LOGGER.isTraceEnabled()) {
-                        LOGGER.trace(format("Checked in connection [%s] to server %s", getId(wrapped), serverId.getAddress()));
-                    }
+                connectionPoolListener.connectionCheckedIn(new ConnectionCheckedInEvent(getId(wrapped)));
+                if (LOGGER.isTraceEnabled()) {
```

Comment:
Specification requires that this event is published even after the pool has been closed.

Reply:
ACK